### ISSUE TITLE
chunkLoadTimeout has to be cleared, if a request for a chunk fails

### ISSUE THREAD

**Issue by artem-malko** [webpack/webpack#14874]
I'm not sure, is it a bug or a feature request. We have an option `chunkLoadTimeout` in output. Here is a link to [the docs](https://webpack.js.org/configuration/output/#outputchunkloadtimeout). `chunkLoadTimeout` is the Number of milliseconds before chunk request expires. But there is a problem, when a request fails. 

Let's imagine, that we have a retry for a dynamic import. If the request for a chunk failed with a 500 code (for example), webpack will wait 12000 ms by default before the retry will be executed, cause import-Promise won't be resolved/rejected during `chunkLoadTimeout`.

So, there is an idea to clearTimeout, if the request failed. What do you think?

**Comment by alexander-akait** [comment]
Please use the issue template

**Comment by artem-malko** [comment]
@alexander-akait updated)

**Comment by sokra** [comment]
> Let's imagine, that we have a retry for a dynamic import. If the request for a chunk failed with a 500 code (for example), webpack will wait 12000 ms by default before the retry will be executed, cause import-Promise won't be resolved/rejected during `chunkLoadTimeout`.

I'm pretty sure it doesn't behave like that. We reject the Promise as soon the script load errors: https://github.com/webpack/webpack/blob/f07a3b82327f4f7894500595f6619c7880eb31a7/lib/runtime/LoadScriptRuntimeModule.js#L150

**Comment by sokra** [comment]
Maybe you can create a repro to demonstrate the incorrect behavior.

**Comment by artem-malko** [comment]
@sokra ok, i'll try to do it on a weekend

**Comment by artem-malko** [comment]
Looks like there is no any problem with a simple js-code. But, if there is a React component imported via 

```js
import('./comp');
```

There is can be a problem. I'll continue to figure out the problem. Stay tuned)

**Comment by artem-malko** [comment]
As I understand, the main problem is that webpack adds onerror handler after script was loaded/failed.

If I add a delay for the dynamic script, everything works perfectly:

```js
app.use((req, res, next) => {
  if (req.url.includes('dynamic')) {
    setTimeout(() => {
      res.status(404).send();
    }, 1000);
  } else {
    next();
  }
});
```

In that case retry starts after 3 seconds (1 second — delay from the server, 2 seconds from LazyLoader).

**Comment by artem-malko** [comment]
By the way, there is a really simple solution =)

```js
<script src="http://localhost:3000/dist/dynamic.js" async onerror="this.remove()"></script>
```

If script is errored before webpack run time start, this script will be removed and webpack starts loading again by itself)

**Comment by alexander-akait** [comment]
> But where can be some cases, when I need to preload a dynamic js file via <script async>. For example, this is a recommended way to preload js-files in React 18. 

Yep, I see... I think your solution with `onerror` is good, I am afraid we just do not put such use in the architecture.

**Comment by artem-malko** [comment]
@alexander-akait should I update the docs?) May be this solution can be useful for somebody.

**Comment by alexander-akait** [comment]
@artem-malko Let's wait @sokra answer too, maybe we will have other ideas too

**Comment by artem-malko** [comment]
@sokra up) We need your opinion)

**Comment by artem-malko** [comment]
@alexander-akait looks like @sokra is not interested in it) What is the next step?)

**Comment by sokra** [comment]
To my knowledge the loading error that happened with the dynamic script is not detectable. We already attach `error` and `load` event listener, but they do not trigger as the error happens too early.

The `onerror="this.remove()"` is a possible solution.

Another solution is to use `<link rel="preload" as="script" href="...">` instead of the script tag.

**Comment by artem-malko** [comment]
@sokra wow! Thx for the replay)

Should I add this info to the docs?

**Comment by alexander-akait** [comment]
I think yes :+1:

**Comment by artem-malko** [comment]
@alexander-akait do you have any idea, how to start with it?) Where should I add such info?

**Comment by alexander-akait** [comment]
You can add explain after options here https://github.com/webpack/webpack.js.org and provide some hints

**Comment by artem-malko** [comment]
@alexander-akait https://github.com/webpack/webpack.js.org/pull/5960 pull-request is ready)
